src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+     * <p/>
+     * This returns the ordinal of the enum {@link DayOfWeek}.
+     * Day-of-week do not match those of the ISO calendar system.
+     * In particular, each months starts with Sunday.
+     * Leap-day and year-day are not considered week-days, thus return 0.
+     *
+     * @return the day of the week enumeration: between 1 (Monday) and 7 (Sunday), or 0 (leap-day, year-day)
+     */
+    @Override
+    public int getDayOfWeek() {
+        if (isLeapDay() || isYearDay()) {
+            return 0;
+        }
+
+        int doy = getDayOfYearAdjusted();
+
+        return 1 + ((5 + doy) % 7);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return
+     */
+    @Override
+    long getProlepticMonth() {
+        return getProlepticYear() * lengthOfYearInMonths() + getMonth() - 1;
+    }
+
+    private long getCalculatedProlepticMonth() {
+        return getProlepticYear() * lengthOfYearInMonths() + getCalculatedMonth() - 1;
+    }
+
+    //-------------------------------------------------------------------------
+    @Override  // for covariant return type
+    @SuppressWarnings ("unchecked")
+    public ChronoLocalDateTime<InternationalFixedDate> atTime(final LocalTime localTime) {
+        return (ChronoLocalDateTime<InternationalFixedDate>) ChronoLocalDate.super.atTime(localTime);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return number of TemporalUnit until given Temporal
+     */
+    @Override
+    public long until(final Temporal endExclusive, final TemporalUnit unit) {
+        InternationalFixedDate end = InternationalFixedDate.from(endExclusive);
+
+        if (unit instanceof ChronoUnit) {
+            switch ((ChronoUnit) unit) {
+                case DAYS:
+                    return daysUntil(end);
+                case WEEKS:
+                    return daysUntil(end) / InternationalFixedChronology.DAYS_IN_WEEK;
+                case MONTHS:
+                    return monthsUntil(end);
+                case YEARS:
+                    return monthsUntil(end) / InternationalFixedChronology.MONTHS_IN_YEAR;
+                case DECADES:
+                    return monthsUntil(end) / (InternationalFixedChronology.MONTHS_IN_YEAR * 10);
+                case CENTURIES:
+                    return monthsUntil(end) / (InternationalFixedChronology.MONTHS_IN_YEAR * 100);
+                case MILLENNIA:
+                    return monthsUntil(end) / (InternationalFixedChronology.MONTHS_IN_YEAR * 1000);
+                case ERAS:
+                    return end.getLong(ChronoField.ERA) - getLong(ChronoField.ERA);
+            }
+
+            throw new UnsupportedTemporalTypeException("Unsupported unit: " + unit);
+        }
+
+        return unit.between(this, end);
+    }
+
+    /**
+     * Get the number of years from this date to the given day.
+     *
+     * @param end The end date.
+     * @return The number of years from this date to the given day.
+     */
+    long yearsUntil(final InternationalFixedDate end) {
```

Comment:
Corrected it now, to be used in the method until() above it.
Au contraire, day-of-year is very important here.